source/base/parameter_handler.cc

```diff
@@ -703,14 +704,29 @@ void
 ParameterHandler::declare_entry(const std::string &          entry,
                                 const std::string &          default_value,
                                 const Patterns::PatternBase &pattern,
-                                const std::string &          documentation)
+                                const std::string &          documentation,
+                                const bool &                 has_to_be_set)
 {
   entries->put(get_current_full_path(entry) + path_separator + "value",
                default_value);
   entries->put(get_current_full_path(entry) + path_separator + "default_value",
                default_value);
   entries->put(get_current_full_path(entry) + path_separator + "documentation",
                documentation);
+  entries->put(get_current_full_path(entry) + path_separator + "has_to_be_set",
+               has_to_be_set);
+  entries->put(get_current_full_path(entry) + path_separator + "has_been_set",
+               false);
```

Comment:
If you do things here in the tree, then you won't need the `entries_set_status` object any more, right? You're just storing the same information twice -- that's a prescription for getting data structures that are out of sync.